saleor/core/utils/__init__.py

```diff
@@ -145,7 +150,11 @@ def prepare_unique_slug(slug: str, slug_values: Iterable):
 
     while unique_slug in slug_values:
         extension += 1
-        unique_slug = f"{slug}-{extension}"
+
+        if slug == "-":
```

Comment:
I might be wrong, but I am not convinced that this will work. 
See for example pattern from line 130. 
If I will have a products with slug: "-1", "-2", "-3":
```python
p = Product.objects.create(slug="-2", product_type_id=10)
p = Product.objects.create(slug="-3", product_type_id=10)
slug = "-"
slug_field_name = "slug"
search_field = f"{slug_field_name}__iregex"
pattern = rf"{slug}-\d+$|{slug}$"
lookup = {search_field: pattern}
slug_values = (
        Product.objects.filter(**lookup)
        .exclude(pk=instance.pk)
        .values_list(slug_field_name, flat=True)
    )
```
```python
slug_values
Out[27]: <ProductsQueryset ['-']>
```

Where the same lookup for `slug` == `3` returns different list:
```python
p = Product.objects.create(slug="3-1", product_type_id=10)
p = Product.objects.create(slug="3-2", product_type_id=10)
p = Product.objects.create(slug="3-3", product_type_id=10)

slug_values
Out[37]: <ProductsQueryset ['3-1', '3-2', 'polo-shirt-3', 't-shirt-3']>
```

Reply:
thanks, I will check it

Reply:
@korycins You were right. I removed line `unique_slug = f"{slug}{extension}"` to keep the pattern we are using so slugs for this case will look like `-`, `--2`, `--3` etc.